dawn/test/unit-test/dawn/IIR/TestIIRSerializer.cpp

```diff
@@ -333,13 +286,62 @@ TEST_F(IIRSerializerTest, IIRTests) {
+  stmt->getData<iir::IIRStmtData>().CallerAccesses = std::make_optional(std::move(stmtAccesses));
   auto stmtAccessPair = std::make_unique<iir::StatementAccessesPair>(stmt);
-  std::shared_ptr<iir::Accesses> callerAccesses = std::make_shared<iir::Accesses>();
-  stmtAccessPair->setCallerAccesses(callerAccesses);
 
   (IIRDoMethod)->insertChild(std::move(stmtAccessPair));
+  std::string varName = "foo";
+  auto varDeclStmt = iir::makeVarDeclStmt(dawn::Type(BuiltinTypeID::Float), varName, 0, "=",
+                                          std::vector<std::shared_ptr<iir::Expr>>{expr->clone()});
+  iir::Accesses varDeclStmtAccesses;
+  varDeclStmtAccesses.addWriteExtent(33, extents);
+  varDeclStmt->getData<iir::IIRStmtData>().CallerAccesses =
+      std::make_optional(std::move(varDeclStmtAccesses));
+  varDeclStmt->getData<iir::VarDeclStmtData>().AccessID = std::make_optional<int>(33);
+  auto varDeclSAP = std::make_unique<iir::StatementAccessesPair>(varDeclStmt);
+
+  (IIRDoMethod)->insertChild(std::move(varDeclSAP));
+
+  deserialized = serializeAndDeserializeRef();
+  IIR_EXPECT_EQ(deserialized, referenceInstantiaton);
+  auto deserializedExprStmt = std::dynamic_pointer_cast<iir::ExprStmt>(deserialized->getIIR()
+                                                                           ->getChild(0)
+                                                                           ->getChild(0)
+                                                                           ->getChild(0)
+                                                                           ->getChild(0)
+                                                                           ->getChild(0)
+                                                                           ->getStatement());
+  deserializedExprStmt->getData<iir::IIRStmtData>().CallerAccesses->addReadExtent(50, extents);
+  IIR_EXPECT_NE(deserialized, referenceInstantiaton);
+  deserialized = serializeAndDeserializeRef();
+  auto deserializedVarAccessExpr = std::dynamic_pointer_cast<iir::VarAccessExpr>(
+      std::dynamic_pointer_cast<iir::ExprStmt>(deserialized->getIIR()
+                                                   ->getChild(0)
+                                                   ->getChild(0)
+                                                   ->getChild(0)
+                                                   ->getChild(0)
+                                                   ->getChild(0)
+                                                   ->getStatement())
+          ->getExpr());
+  deserializedVarAccessExpr->getData<iir::IIRAccessExprData>().AccessID =
+      std::make_optional<int>(50);
+  IIR_EXPECT_NE(deserialized, referenceInstantiaton);
+  deserialized = serializeAndDeserializeRef();
+  auto deserializedVarDeclStmt = std::dynamic_pointer_cast<iir::VarDeclStmt>(deserialized->getIIR()
+                                                                                 ->getChild(0)
```

Comment:
that looks very verbose... is this really what we want to do here?

Reply:
I think testing data serialization is nice to have ... but I couldn't come up with a shorter way to write that ugly chain: `getIIR()
                                                                                 ->getChild(0)
                                                                                 ->getChild(0)
                                                                                 ->getChild(0)
                                                                                 ->getChild(0)
                                                                                 ->getChild(1)
                                                                                 ->getStatement()`

Reply:
I've come up with this to get the first child: `iterateIIROver<iir::StatementAccessesPair>(deserialized->getIIR()).begin()` (`++` for the second) ... not sure if improves readability

Reply:
No it's okay, I was just wondering whether this was intentional. What might improve readability, is to do it in several steps:
```
auto xyz = getIIR()->getChild(0);
auto abc = xyz->getChild(0);
...
```
because then, it would be clear what all these `getChild` are. What do you think?

Reply:
yes can do